src/main/java/com/metamx/emitter/core/HttpEmitterConfig.java

```diff
@@ -17,258 +17,47 @@
-      long flushMillis,
-      int flushCount,
-      String recipientBaseUrl,
-      String basicAuthentication,
-      BatchingStrategy batchingStrategy,
-      int maxBatchSize,
-      long maxBufferSize
-  )
-  {
-    this(
-        flushMillis,
-        flushCount,
-        DEFAULT_FLUSH_TIME_OUT,
-        recipientBaseUrl,
-        basicAuthentication,
-        batchingStrategy,
-        maxBatchSize,
-        maxBufferSize
-    );
-  }
-
-  /**
-   * @deprecated use {@link Builder}
-   */
-  @Deprecated
-  public HttpEmitterConfig(
-      long flushMillis,
-      int flushCount,
-      long flushTimeOut,
-      String recipientBaseUrl,
-      String basicAuthentication,
-      BatchingStrategy batchingStrategy,
-      int maxBatchSize,
-      long maxBufferSize
-  )
-  {
-    this(
-        flushMillis,
-        flushCount,
-        flushTimeOut,
-        recipientBaseUrl,
-        basicAuthentication,
-        batchingStrategy,
-        maxBatchSize,
-        maxBufferSize,
-        DEFAULT_CONTENT_ENCODING
-    );
-  }
-
-  private HttpEmitterConfig(
-      long flushMillis,
-      int flushCount,
-      long flushTimeOut,
-      String recipientBaseUrl,
-      String basicAuthentication,
-      BatchingStrategy batchingStrategy,
-      int maxBatchSize,
-      long maxBufferSize,
-      ContentEncoding contentEncoding
-  )
+  public HttpEmitterConfig(BaseHttpEmittingConfig base, String recipientBaseUrl)
   {
-    this.flushMillis = flushMillis;
-    this.flushCount = flushCount;
-    this.flushTimeOut = flushTimeOut;
+    this.flushMillis = base.flushMillis;
+    this.flushCount = base.flushCount;
+    this.flushTimeOut = base.flushTimeOut;
     this.recipientBaseUrl = recipientBaseUrl;
-    this.basicAuthentication = basicAuthentication;
-    this.batchingStrategy = batchingStrategy;
-    this.maxBatchSize = maxBatchSize;
-    this.maxBufferSize = maxBufferSize;
-    this.contentEncoding = contentEncoding;
-  }
-
-  public long getFlushMillis()
-  {
-    return flushMillis;
-  }
-
-  public int getFlushCount()
-  {
-    return flushCount;
-  }
-
-  public long getFlushTimeOut() {
-    return flushTimeOut;
+    this.basicAuthentication = base.basicAuthentication;
+    this.batchingStrategy = base.batchingStrategy;
+    this.maxBatchSize = base.maxBatchSize;
+    this.maxBufferSize = base.maxBufferSize;
+    this.contentEncoding = base.contentEncoding;
   }
 
   public String getRecipientBaseUrl()
   {
     return recipientBaseUrl;
   }
 
-  public String getBasicAuthentication()
-  {
-    return basicAuthentication;
-  }
-
-  public BatchingStrategy getBatchingStrategy()
-  {
-    return batchingStrategy;
-  }
-
-  public int getMaxBatchSize()
-  {
-    return maxBatchSize;
-  }
-
-  public long getMaxBufferSize()
-  {
-    return maxBufferSize;
-  }
-
-  public ContentEncoding getContentEncoding() {
-    return contentEncoding;
-  }
-
-  public static class Builder
+  public static class Builder extends HttpEmitterConfig
```

Comment:
extending HttpEmitterConfig is good to avoid code duplication but can lead to errors when somebody forget to call `build()` in the end and it won't be caught by compiler.